engine_wrapper.py

```diff
@@ -316,41 +330,36 @@ def get_stats(self, for_chat: bool = False) -> list[str]:
         can_index = self.move_commentary and self.move_commentary[-1]
         info: MOVE_INFO_TYPE = self.move_commentary[-1].copy() if can_index else {}
 
-        def to_readable_value(stat: str, info: MOVE_INFO_TYPE) -> str:
-            readable: dict[str, Callable[[Any], str]] = {"score": self.readable_score, "wdl": self.readable_wdl,
-                                                         "hashfull": lambda x: f"{round(x / 10, 1)}%",
-                                                         "nodes": self.readable_number,
-                                                         "nps": lambda x: f"{self.readable_number(x)}nps",
-                                                         "tbhits": self.readable_number,
-                                                         "cpuload": lambda x: f"{round(x / 10, 1)}%"}
-
-            def identity(x: Any) -> str:
-                return str(x)
-
-            return str(readable.get(stat, identity)(info[stat]))
-
-        def to_readable_key(stat: str) -> str:
+        def to_readable_key(stat: str, value: Any) -> str:
             readable = {"wdl": "winrate", "ponderpv": "PV", "nps": "speed", "score": "evaluation"}
             stat = readable.get(stat, stat)
+            if stat == "string" and value in ["Opening Book", "Lichess Opening Explorer (Masters)",
+                                              "Lichess Opening Explorer (Player)", "Lichess Opening Explorer (Lichess)",
+                                              "Lichess EGTB", "ChessDB EGTB", "Lichess Cloud Analysis", "ChessDB"]:
+                stat = "source"
             return stat.title()
 
-        stats = ["score", "wdl", "depth", "nodes", "nps", "ponderpv"]
-        if for_chat and "ponderpv" in info:
-            bot_stats = [f"{to_readable_key(stat)}: {to_readable_value(stat, info)}"
-                         for stat in stats if stat in info and stat != "ponderpv"]
+        info = {to_readable_key(key, value): value for (key, value) in info.items()}
+        if "Source" not in info:
+            info["Source"] = "Engine"
```

Comment:
This should be changed to this:
```py
        info = {to_readable_key(stat, value): value for (stat, value) in info.items()}
        if "Source" not in info:
            info["Source"] = "Engine"
```
Otherwise the `info["Source"]` field will always be `"Engine"`

Reply:
How does changing the name of the variable from `key` to `stat` change anything?

Reply:
I was testing this code and I found 2 errors:
1. Source is always "Engine", which gets fixed by my previous comment.
2. `to_readable_value` doesn't really seem to execute properly. My stats are printed like this:
```
[08/07/23 10:27:31] INFO     Source: Engine                engine_wrapper.py:233
                    INFO     Evaluation: PovScore(Cp(0),   engine_wrapper.py:233
                             WHITE)                                             
                    INFO     Depth: 51                     engine_wrapper.py:233
                    INFO     Nodes: 22616579               engine_wrapper.py:233
                    INFO     Speed: 1360313                engine_wrapper.py:233
                    INFO     Pv: 60. Rg8+ Kh5 61. Re5+ Kh6 engine_wrapper.py:233
                             62. Rge8 Kg6 63. R5e7 Kh5 64.                      
                             Rh8+ Kg6 65. Rg8+ Kh6 66.                          
                             Rh8+
```